### PR TITLE
PERF-#287: Optimize python list appends 

### LINKED ISSUE #287: Optimize python code snippets in MPI backend

### PR DESCRIPTION
**Pull request by arunjose696**
## What do these changes do?

- [ ] first commit message and PR title follow format outlined [here](https://unidist.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.

Computed time with 
```
elapsed_time=0   
for i in range(3):
    start = time.time()
    output_ids = object_store.generate_output_data_id(
        1, garbage_collector, 1000000
    )
    end = time.time()
    elapsed_time+= (end-start)
print(elapsed_time/3)
```

Slight improvement on current(4.316426235834758) vs master (4.434067408243815), The time improvement not significant when running benchmarks

- [ ] passes `flake8 .`
- [ ] passes `black .`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #287
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: unidist/core/backends/mpi/core/controller/object_store.py [hunk 1]
```diff
@@ -196,10 +196,10 @@ def generate_output_data_id(self, dest_rank, gc, num_returns=1):
         elif num_returns == 0:
             output_ids = None
         else:
-            output_ids = []
-            for _ in range(num_returns):
+            output_ids = [None] * num_returns
```
Comment (YarShev) [inline unidist/core/backends/mpi/core/controller/object_store.py:199]: We use many .append calls in communcation, for instance. Let's fix all the places. Also, please take a look at other python code snippets that might be improved.

Comment (YarShev) [reply unidist/core/backends/mpi/core/controller/object_store.py:199]: > We use many .append calls in communcation, for instance. Let's fix all the places

Of course, if it gives us benefit.

Comment (arunjose696) [reply unidist/core/backends/mpi/core/controller/object_store.py:199]: Could find only 2 appends that could be replaced with inserting at list indexes, 

Few other places use conditions to check and then append having a python list of fixed size would not work for these cases , tried list comprehensions with ":=" operator for these as in this commit
https://github.com/modin-project/unidist/pull/290/commits/d965f071aa4eb30893b1b1d354321329f499d500#diff-38853ac231026c17efa13d47a868b3d3ef66be4d2e29e3e2ddc58539423b422eR92
But python<3.8 does not support ":=" so removed those.

Comment (YarShev) [reply unidist/core/backends/mpi/core/controller/object_store.py:199]: One of the examples in communication. https://github.com/modin-project/unidist/blob/edbaadf80d5911d3723268950a42b776282448dc/unidist/core/backends/mpi/core/communication.py#L272

Comment (YarShev) [reply unidist/core/backends/mpi/core/controller/object_store.py:199]: Please try to replace appends to list indexing check the perf once again on our benchmarks.

**Review by YarShev — APPROVED** [review]

**Pull request merged**